replication/binlogsyncer.go

```diff
@@ -635,9 +627,15 @@ func (b *BinlogSyncer) onStream(s *BinlogStreamer) {
 
 	for {
 		data, err := b.c.ReadPacket()
+		select {
```

Comment:
I suggest don't add this logic, because we don't know which event (`fail to read packet` , `ctx canceled`) happen firstly

Reply:
Without this, calling `canal.Close()` always results in a `failed to read packet` error.

More generally, what I'm trying to do is make clean shutdown (no errors) possible. I'm open to alternative ideas.